webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/HomePage.scala

```diff
@@ -421,7 +421,7 @@ class HomePage extends Loggable {
       }.toMap.mapValues{_.maxBy(_.value)} // Get the max version available
 
       // take back the initial set of nodes to be sure to have one agent for each
-      val allAgents = nodeInfos.keySet.toSeq.map(nodeId => agentVersionByNodeEntries.get(nodeId) match {
+      val allAgents = nodeInfos.keysIterator.toSeq.map(nodeId => agentVersionByNodeEntries.get(nodeId) match {
```

Comment:
maybe overkill, as it was a keySet.toSeq.map

Reply:
`keySet.toSet.map` is less effecient than `keysIterator.map` (the seq is not created)